[Bug]: core panic in getPositionMetricTotal during vega-market-sim fuzzing runs

### Problem encountered

VMS [PR](https://github.com/vegaprotocol/vega-market-sim/pull/526) narrows fuzzing run to focus on"valid" recurring transfers funding reward pools.

Core panics during market-sim fuzzing runs.

[vega-sim-zfonk035.zip](https://github.com/vegaprotocol/vega/files/12892297/vega-sim-zfonk035.zip)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0xe8 pc=0x105458810]

goroutine 45103 [running]:
code.vegaprotocol.io/vega/core/execution/common.(*marketTracker).getPositionMetricTotal(...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:970
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForParty(0x14000ee0800, {0x14008586000, 0x40}, {0x1400350f0c0, 0x40}, {0x14008577930, 0x1, 0x140096cb8c0?}, 0x5, 0x2)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:716 +0x4b0
code.vegaprotocol.io/vega/core/execution/common.calculateMetricForTeamUtil({0x14008586000, 0x40}, {0x140096acbd0?, 0x1, 0x1400e47ea50?}, {0x14008577930, 0x1, 0x1}, 0x30?, 0x107e5b8e0?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:653 +0x16c
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeam(0x10796a660?, {0x14008586000?, 0x1400350f3c0?}, {0x140096acbd0?, 0x14007f79b48?, 0x1043953c8?}, {0x14008577930?, 0x0?, 0x14007f79b68?}, 0x434b84c?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:634 +0x98
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeams(0x14008438468?, {0x14008586000, 0x40}, 0x1400e47ea20, {0x14008577930, 0x1, 0x1}, 0x7950800?, 0x1?, 0x14007f79d08?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:621 +0x298
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).CalculateMetricForTeams(0x14000ee0800, 0x14008548540)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:577 +0x14c
code.vegaprotocol.io/vega/core/banking.(*Engine).dispatchRequired(0x1400060c4e0?, 0x14007f7a090?)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:206 +0x74
code.vegaprotocol.io/vega/core/banking.(*Engine).distributeRecurringTransfers(0x1400060c4e0, {0x108036578, 0x1400cc02900}, 0x14)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:302 +0x1894
code.vegaprotocol.io/vega/core/banking.(*Engine).OnEpoch(0x1400060c4e0, {0x108036578, 0x1400cc02900}, {0x14, {0x0, 0xedbec19b0, 0x0}, {0x0, 0xedbec1a28, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/banking/engine.go:280 +0xb4
code.vegaprotocol.io/vega/core/epochtime.(*Svc).notify(0x140016b65a0, {0x108036578, 0x1400cc02900}, {0x14, {0x0, 0xedbec19b0, 0x0}, {0x0, 0xedbec1a28, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:204 +0xec
code.vegaprotocol.io/vega/core/epochtime.(*Svc).OnTick(0x140016b65a0, {0x108036578, 0x1400cc02900}, {0x1?, 0x108050d88?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:134 +0x58c
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x14000ee0680, {0x108036578, 0x1400cc02900}, {0x40?, 0x1077b8fa0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(_, {{0x140091ebce0, 0x20, 0x20}, {{0x0, 0x0}, {0x14000518d28, 0x6}, 0x98b, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:840 +0x5bc
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).BeginBlock(_, {{0x140091ebce0, 0x20, 0x20}, {{0x0, 0x0}, {0x14000518d28, 0x6}, 0x98b, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:49 +0x64
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).BeginBlock(_, {{0x140091ebce0, 0x20, 0x20}, {{0x0, 0x0}, {0x14000518d28, 0x6}, 0x98b, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:67 +0x4c
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).BeginBlock(0x1400023e000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:340 +0x2e4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x1400023e000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:199 +0x1b8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x1400023e000, 0x3b9aca00)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:276 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec
```

### Observed behaviour

Panic

### Expected behaviour

No panic

### Steps to reproduce

```bash
1. cd vega-market-sim
1. git checkout feat/rework_reward_fuzzing
1. python -m vega_sim.replay.replay --dir PATH_TO_REPLAY_FILE
```


### Software version

529d5dfa85cbd9329fbc5ff4a869630577e77eac

### Failing test

_No response_

### Jenkins run

_No response_

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "VOTE": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "VOTE",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "100",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0xe8 pc=0x105458810]

goroutine 45103 [running]:
code.vegaprotocol.io/vega/core/execution/common.(*marketTracker).getPositionMetricTotal(...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:970
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForParty(0x14000ee0800, {0x14008586000, 0x40}, {0x1400350f0c0, 0x40}, {0x14008577930, 0x1, 0x140096cb8c0?}, 0x5, 0x2)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:716 +0x4b0
code.vegaprotocol.io/vega/core/execution/common.calculateMetricForTeamUtil({0x14008586000, 0x40}, {0x140096acbd0?, 0x1, 0x1400e47ea50?}, {0x14008577930, 0x1, 0x1}, 0x30?, 0x107e5b8e0?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:653 +0x16c
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeam(0x10796a660?, {0x14008586000?, 0x1400350f3c0?}, {0x140096acbd0?, 0x14007f79b48?, 0x1043953c8?}, {0x14008577930?, 0x0?, 0x14007f79b68?}, 0x434b84c?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:634 +0x98
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeams(0x14008438468?, {0x14008586000, 0x40}, 0x1400e47ea20, {0x14008577930, 0x1, 0x1}, 0x7950800?, 0x1?, 0x14007f79d08?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:621 +0x298
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).CalculateMetricForTeams(0x14000ee0800, 0x14008548540)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:577 +0x14c
code.vegaprotocol.io/vega/core/banking.(*Engine).dispatchRequired(0x1400060c4e0?, 0x14007f7a090?)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:206 +0x74
code.vegaprotocol.io/vega/core/banking.(*Engine).distributeRecurringTransfers(0x1400060c4e0, {0x108036578, 0x1400cc02900}, 0x14)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:302 +0x1894
code.vegaprotocol.io/vega/core/banking.(*Engine).OnEpoch(0x1400060c4e0, {0x108036578, 0x1400cc02900}, {0x14, {0x0, 0xedbec19b0, 0x0}, {0x0, 0xedbec1a28, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/banking/engine.go:280 +0xb4
code.vegaprotocol.io/vega/core/epochtime.(*Svc).notify(0x140016b65a0, {0x108036578, 0x1400cc02900}, {0x14, {0x0, 0xedbec19b0, 0x0}, {0x0, 0xedbec1a28, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:204 +0xec
code.vegaprotocol.io/vega/core/epochtime.(*Svc).OnTick(0x140016b65a0, {0x108036578, 0x1400cc02900}, {0x1?, 0x108050d88?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:134 +0x58c
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x14000ee0680, {0x108036578, 0x1400cc02900}, {0x40?, 0x1077b8fa0?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(_, {{0x140091ebce0, 0x20, 0x20}, {{0x0, 0x0}, {0x14000518d28, 0x6}, 0x98b, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:840 +0x5bc
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).BeginBlock(_, {{0x140091ebce0, 0x20, 0x20}, {{0x0, 0x0}, {0x14000518d28, 0x6}, 0x98b, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:49 +0x64
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).BeginBlock(_, {{0x140091ebce0, 0x20, 0x20}, {{0x0, 0x0}, {0x14000518d28, 0x6}, 0x98b, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:67 +0x4c
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).BeginBlock(0x1400023e000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:340 +0x2e4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x1400023e000)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:199 +0x1b8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x1400023e000, 0x3b9aca00)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:276 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec
```

@ze97286 different panic on `c5b31b076f8e3f99d08d0fb3585a0371f34c8be5 ` now in `getReturns` - should I raise a separate issue?
[vega-sim-5j1855vq.zip](https://github.com/vegaprotocol/vega/files/12892581/vega-sim-5j1855vq.zip)

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x103fd661c]

goroutine 36034 [running]:
code.vegaprotocol.io/vega/core/execution/common.(*marketTracker).getReturns(0x0, {0x14003441840, 0x40}, 0x2)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:949 +0x2c
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForParty(0x1400012c980, {0x1400c514d00, 0x40}, {0x14003441840, 0x40}, {0x14006a56e50, 0x1, 0x1400977cb40?}, 0x7, 0x2)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:720 +0x528
code.vegaprotocol.io/vega/core/execution/common.calculateMetricForTeamUtil({0x1400c514d00, 0x40}, {0x140096dff90?, 0x1, 0x1400c2a2ff0?}, {0x14006a56e50, 0x1, 0x1}, 0x30?, 0x1069d77e0?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:653 +0x16c
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeam(0x1064e6520?, {0x1400c514d00?, 0x14003441b40?}, {0x140096dff90?, 0x1400a6d1b48?, 0x102f10b88?}, {0x14006a56e50?, 0x0?, 0x1400a6d1b68?}, 0x2ec700c?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:634 +0x98
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).calculateMetricForTeams(0x1400688d338?, {0x1400c514d00, 0x40}, 0x1400c2a2fc0, {0x14006a56e50, 0x1, 0x1}, 0x64cc6c0?, 0x1?, 0x1400a6d1d08?, ...)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:621 +0x298
code.vegaprotocol.io/vega/core/execution/common.(*MarketActivityTracker).CalculateMetricForTeams(0x1400012c980, 0x140068b97a0)
	/Users/caed/vega-market-sim/extern/vega/core/execution/common/market_activity_tracker.go:577 +0x14c
code.vegaprotocol.io/vega/core/banking.(*Engine).dispatchRequired(0x14001776820?, 0x1400a6d2090?)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:206 +0x74
code.vegaprotocol.io/vega/core/banking.(*Engine).distributeRecurringTransfers(0x14001776820, {0x106bb2658, 0x1400b769500}, 0x13)
	/Users/caed/vega-market-sim/extern/vega/core/banking/recurring_transfers.go:302 +0x1894
code.vegaprotocol.io/vega/core/banking.(*Engine).OnEpoch(0x14001776820, {0x106bb2658, 0x1400b769500}, {0x13, {0x0, 0xedbec1936, 0x0}, {0x0, 0xedbec19ae, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/banking/engine.go:280 +0xb4
code.vegaprotocol.io/vega/core/epochtime.(*Svc).notify(0x14001774480, {0x106bb2658, 0x1400b769500}, {0x13, {0x0, 0xedbec1936, 0x0}, {0x0, 0xedbec19ae, 0x0}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:204 +0xec
code.vegaprotocol.io/vega/core/epochtime.(*Svc).OnTick(0x14001774480, {0x106bb2658, 0x1400b769500}, {0x1?, 0x106bcce68?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/epochtime/service.go:134 +0x58c
code.vegaprotocol.io/vega/core/vegatime.(*Svc).notify(...)
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:107
code.vegaprotocol.io/vega/core/vegatime.(*Svc).SetTimeNow(0x1400012c800, {0x106bb2658, 0x1400b769500}, {0x40?, 0x106334e60?, 0x0?})
	/Users/caed/vega-market-sim/extern/vega/core/vegatime/service.go:80 +0x26c
code.vegaprotocol.io/vega/core/processor.(*App).OnBeginBlock(_, {{0x14009660b00, 0x20, 0x20}, {{0x0, 0x0}, {0x1400139dca0, 0x6}, 0x911, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/processor/abci.go:840 +0x5bc
code.vegaprotocol.io/vega/core/blockchain/abci.(*App).BeginBlock(_, {{0x14009660b00, 0x20, 0x20}, {{0x0, 0x0}, {0x1400139dca0, 0x6}, 0x911, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/abci/abci.go:49 +0x64
code.vegaprotocol.io/vega/cmd/vega/commands/node.(*appW).BeginBlock(_, {{0x14009660b00, 0x20, 0x20}, {{0x0, 0x0}, {0x1400139dca0, 0x6}, 0x911, {0x0, ...}, ...}, ...})
	/Users/caed/vega-market-sim/extern/vega/cmd/vega/commands/node/app_wrapper.go:67 +0x4c
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).BeginBlock(0x14001774c60)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:340 +0x2e4
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).processBlock(0x14001774c60)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:199 +0x1b8
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).ForwardTime(0x14001774c60, 0x77359400)
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/nullchain.go:276 +0x198
code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime.func1()
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:132 +0x2c
created by code.vegaprotocol.io/vega/core/blockchain/nullchain.(*NullBlockchain).handleForwardTime
	/Users/caed/vega-market-sim/extern/vega/core/blockchain/nullchain/server.go:131 +0x1ec

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -464,6 +464,7 @@
 - [8702](https://github.com/vegaprotocol/vega/issues/8702) - Fix panic on auction exit after stop orders expired in auction
 - [8703](https://github.com/vegaprotocol/vega/issues/8703) - Wire stop orders cancellation
 - [8698](https://github.com/vegaprotocol/vega/issues/8698) - Always set the `ToAccount` field when clearing fees.
+- [9773](https://github.com/vegaprotocol/vega/issues/9773) - Ensure invalid market for transfer doesn't panic
 
 ## 0.71.0
 

--- a/core/execution/common/market_activity_tracker.go
+++ b/core/execution/common/market_activity_tracker.go
@@ -711,6 +711,9 @@ func (mat *MarketActivityTracker) calculateMetricForParty(asset, party string, m
 	// for each market in scope, for each epoch in the time window get the metric entry, sum up for each epoch in the time window and divide by window size (or calculate variance - for volatility)
 	for _, market := range markets {
 		marketTracker := assetTrackers[market]
+		if marketTracker == nil {
+			continue
+		}
 		switch metric {
 		case vega.DispatchMetric_DISPATCH_METRIC_AVERAGE_POSITION:
 			uTotal += marketTracker.getPositionMetricTotal(party, windowSize)

